walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,96 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> BusyTimes = new ArrayList<TimeRange>();
+    Collection<TimeRange> FreeTimes = new ArrayList<TimeRange>();
+    int start_of_day = TimeRange.START_OF_DAY;
+    int eod = TimeRange.END_OF_DAY;
+    int prev_end_time = 0; // End time of the previous event accessed
+
+    // Inital checks for input validity(no attendees ,time is more than 24 hours)
+    if (request.getAttendees().isEmpty()) {
+      FreeTimes.add(TimeRange.fromStartEnd(start_of_day, eod, true));
+      return FreeTimes;
+    }
+
+    if (request.getDuration() > 24 * 60) {
+      return FreeTimes;
+    }
+
```

Comment:
You probably want to sort `events` before you execute your first `for` loop, because your logic depends on each `event`'s `.getWhen().start()` being in ascending order.

Reply:
I will try and address this in another pr later, its taking me a little while to figure this out right now and I want to dedicate most of my time to prepping for our capstone project